Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -147,10 +171,18 @@ internal class ConnectionPool {
             case let .open(pool):
                 self.state = .closing(pool: pool)
             case let .opening(future):
-                let pool = try future.wait()
-                self.state = .closing(pool: pool)
+                do {
+                    let pool = try future.wait()
+                    self.state = .closing(pool: pool)
+                } catch {
+                    self.state = .closed
```

Comment:
I wasn't sure whether we should also throw the error here in addition to transitioning the state. currently if you did something like this you would never see the error:
```swift
let client = try MongoClient("uri-triggering-the-problem-here")
try client.syncClose()
```

but if you tried to use the client at all in between creation and close, you would see it.

ultimately a case like this seemed unlikely to me - why create a client to never use it? - and propagating the error would have complicated client shutdown logic (would we consider the client to have failed closing if its pool failed to open?) so I went with this for now, but let me know what y'all think

Reply:
yeah, I agree that a failure to open the pool doesn't necessarily indicate a failure to close the pool (and thereby the client) -- this seems fine to me